test/libyul/Common.cpp

```diff
@@ -97,9 +97,9 @@ yul::Block yul::test::disambiguate(std::string const& _source)
 	return std::get<Block>(Disambiguator(defaultDialect(), *result.second, {})(result.first->root()));
 }
 
-std::string yul::test::format(std::string const& _source)
+std::string yul::test::format(std::string const& _source, Dialect const& _dialect)
 {
-	return yul::AsmPrinter()(parse(_source).first->root());
+	return yul::AsmPrinter(_dialect)(parse(_source).first->root());
```

Comment:
No point accepting the dialect from the outside here. The dialect used for printing must match the one the AST was parsed with, so:

```suggestion
	Dialect const& dialect = languageToDialect(
		YulStack::Language::StrictAssembly,
		solidity::test::CommonOptions::get().evmVersion(),
		solidity::test::CommonOptions::get().eofVersion()
	);
	return yul::AsmPrinter(dialect)(parse(_source,).first->root());
```

We should really be storing it and passing it around together with the AST.